### ISSUE TITLE
Warnings and errors that are not there.

### ISSUE THREAD

**Issue by alesf** [AtomLinter/linter-phpcs#218]
phpcs v 2.7.1., PSR2

It constantly shows same errors.
It says that class is missing opening or closing brace (which is not) and then it thinks all of the functions are indented incorrectly and it assumes the file is declaring classes and executes logic at the same time.

`<?php` - PSR1.Files.SideEffercts.FoundWithSymbols
`class X extends C` - PSR2.Classes.ClassDeclaration.MissingBrace
`public function __construct()` - Generic.WhiteSpace.ScopeIndent.IncorrectExact

![screen shot 2016-12-12 at 11 12 03](https://cloud.githubusercontent.com/assets/1148574/21096528/997d1d18-c060-11e6-8c61-11b4651c71b5.png)

If I then save the file once or twice to redo the phpcs action it does not show any errors.
And this happens constantly on all the files.

If I run phpcs in console on the same file it does not show any errors.

It's annoying.

Please help.

**Comment by Arcanemagus** [comment]
If messages are "sticking" then either:
* `linter-phpcs` hasn't been called again yet to update its messages
* There is a bug in the Linter package

To verify it isn't the first one save the file and don't make any changes till it is done linting, if the file is changed between when the lint was triggered and the results come back from `phpcs` then those results are discarded as they aren't valid for the current state of the document. Normally this shouldn't be an issue though as the edit that caused the results to be discarded should trigger its own lint call.

**Comment by alesf** [comment]
I took some time to test it.
It seems that these false lint errors usually show up when I modify a file and save it just after changing it. The same happens if I save unmodified file two or three times in a row. I'm not sure about this, but I think these errors mostly show in larger files with 200+ lines.

I'll disable "lint as you type" for now to see if it helps.

Thanks for your suggestions.

**Comment by Arcanemagus** [comment]
Marking this as closed, if there is a bug here it's in Linter.

**Comment by mundschenk-at** [comment]
I've noticed what looks like the same bug. I've a custom coding standard (overriding some WordPress rules) configured in phpcs.xml in the project root. In Principle, that works as expected - except that some errors get triggered from atom that (correctly) don't triggered on the command line.

The most obnoxious of those is `WordPressFilesFileName.InvalidClassFileName` stating that `class-some-name.php` should really be named `class-some-name.php`. To check for the Linter error @Arcanemagus suggested, I disabled "lint as you type" and triggered the linting with single "Save": Again the error appears, apparently in all class files, even minimal ones.
